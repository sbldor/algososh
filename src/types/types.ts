import { ElementStates } from '../types/element-states';

export interface ICharacter {
  character: string
  state: ElementStates
}

export interface IArr {
  num: number
  state: ElementStates
};