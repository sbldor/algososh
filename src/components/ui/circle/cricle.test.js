import renderer from 'react-test-renderer'
import { Circle } from './circle'
import { ElementStates } from '../../../types/element-states'

describe('Circle component snapshot tests', () => {

  test('Circle without text', () => {
    const cir = renderer.create(<Circle />).toJSON()
    expect(cir).toMatchSnapshot()
  })

  test('Circle with letter', () => {
    const cir = renderer.create(<Circle letter='test' />).toJSON()
    expect(cir).toMatchSnapshot()
  })

  test('Circle with head', () => {
    const cir = renderer.create(<Circle head='test' />).toJSON()
    expect(cir).toMatchSnapshot()
  })

  test('Circle with tail', () => {
    const cir = renderer.create(<Circle tail='test' />).toJSON()
    expect(cir).toMatchSnapshot()
  })

  test('Circle with component in head', () => {
    const cir = renderer.create(<Circle head={<Circle />} />).toJSON()
    expect(cir).toMatchSnapshot()
  })

  test('Circle with component in tail', () => {
    const cir = renderer.create(<Circle tail={<Circle />} />).toJSON()
    expect(cir).toMatchSnapshot()
  })

  test('Circle with index', () => {
    const cir = renderer.create(<Circle index={1} />).toJSON()
    expect(cir).toMatchSnapshot()
  })

  test('Circle with isSmall prop', () => {
    const cir = renderer.create(<Circle isSmall />).toJSON()
    expect(cir).toMatchSnapshot()
  })

  test('Circle with Default state', () => {
    const cir = renderer.create(<Circle state={ElementStates.Default} />).toJSON()
    expect(cir).toMatchSnapshot()
  })

  test('Circle with Changing state', () => {
    const cir = renderer.create(<Circle state={ElementStates.Changing} />).toJSON()
    expect(cir).toMatchSnapshot()
  })

  test('Circle with Default state', () => {
    const cir = renderer.create(<Circle state={ElementStates.Modified} />).toJSON()
    expect(cir).toMatchSnapshot()
  })

})