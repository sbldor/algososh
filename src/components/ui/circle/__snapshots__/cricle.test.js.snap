// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component snapshot tests Circle with Changing state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component snapshot tests Circle with Default state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component snapshot tests Circle with Default state 2`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component snapshot tests Circle with component in head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component snapshot tests Circle with component in tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Circle component snapshot tests Circle with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    test
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component snapshot tests Circle with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle component snapshot tests Circle with isSmall prop 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component snapshot tests Circle with letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      test
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component snapshot tests Circle with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    test
  </div>
</div>
`;

exports[`Circle component snapshot tests Circle without text 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
